core/services/pipeline/runner.go

```diff
@@ -187,6 +186,8 @@ func (r *runner) processTaskRun() (anyRemaining bool, err error) {
 	defer cancel()
 
 	return r.orm.ProcessNextUnclaimedTaskRun(ctx, func(ctx context.Context, txdb *gorm.DB, jobID int32, taskRun TaskRun, predecessors []TaskRun) Result {
+		// Note if the context is cancelled while this is running, the tx will error when used.
+		// To avoid confusion we shouldn't log db errors here and instead just wrap the error we return with info.
```

Comment:
I dont think this will work. I think that the transaction context _must_ be separate, or we cannot persist the result.

Reply:
fixed and added a test